LEAF_Request_Portal/sources/Form.php

```diff
@@ -1968,26 +1967,35 @@ public function getEmpUID($userName){
         return $response[0]["empUID"];
     }
 
-    public function checkIfBackup($empUID){
+    /**
+     * checkIfBackup determines if the current user is a backup of the provided $empUID
+     *
+     * @param string $empUID empUID to check
+     * @return boolean
+     */
+    public function checkIfBackup(string|int $empUID): bool
+    {
+        $empUID = (int)$empUID;
+
+        if(isset($this->cache['checkIfBackup'])) {
+            return isset($this->cache['checkIfBackup'][$empUID]);
+        }
 
         $nexusDB = $this->login->getNexusDB();
-        $vars = array(':empId' => $empUID);
-        $backupIds = $nexusDB->prepared_query('SELECT * FROM relation_employee_backup WHERE empUID =:empId', $vars);
 
-        if ($empUID != $this->login->getEmpUID())
-        {
-            foreach ($backupIds as $row)
-            {
-                if ($row['backupEmpUID'] == $this->login->getEmpUID())
-                {
-                    return true;
-                }
-            }
+        $vars = array(':currEmpUID' => $this->login->getEmpUID());
+        $strSQL = 'SELECT empUID FROM relation_employee_backup
+                    WHERE backupEmpUID =:currEmpUID
+                        AND approved=1';
+        $backupIds = $nexusDB->prepared_query($strSQL, $vars);
 
-            return false;
+        $this->cache['checkIfBackup'] = [];
+        foreach ($backupIds as $row)
+        {
+            $this->cache['checkIfBackup'][$row['empUID']] = true;
         }
 
-        return true;
+        return isset($this->cache['checkEmployeeAccess'][$empUID]);
```

Comment:
Note issue with wrong variable here fixed in https://github.com/department-of-veterans-affairs/LEAF/pull/2178